### DISCUSSION TITLE
Selling travel passes via OSDM

### DISCUSSION THREAD

**Discussion by mvdgraaff** [UnionInternationalCheminsdeFer/OSDM#765] [Ideas]
In the meeting of 2024/11/22 we talked about selling travelpasses by using the OSDM specification. In the presentation the following topics were highlighted.
### Selling Passes
1. Flexibility during purchase
    a. Customers want to travel but don’t know where to go. Just an idea of one or more countries.
    b. Customers travel when they want (activation of induvidual travel days)
    c. Optional reservations when needed, last minute decision by the customer
2. Carriers rules
    a. EU / NoN EU citizin (inbound / outbound travel) (Different product portofolio)
    b. Cannot compete with  point 2 point tickets
3. Regional passes instead of p2p tickets like Discoverholland and Euregio ticket (NS)

### OSDM Challenges with passes.
1. OSDM works with the concept of an “offer request”
   a. tripSpecification (Customer doesn’t know where he want to go)
   b. tripSearchCriteria (Customer doesn’t know where he want to go)
   c. nonTripSearchCriteria (places, for OCP which country to visit)
   d. offerSearchCriteria (travelClass and ProductTags {EU|INT, GP|OCP, 1|2|3m, 4|5|7d)
   e. anonymousPassengers (age, dob, residency)
3. Selling passes is more like a product shop approach 
  a. Shopping cart idea
  b. Customer select a pass and buys it. Therefore he needs to know the price. 
4. Passes have different rules, like which carriers can be used, only in a region.

Questions that surface are:
Q1. Does the products api need more get operations?
Q2. What is the best way to request a pass offer?
Q3 Are the elements in offer sufficient?
Q4. How do I sell a reservation to pass holder?
Q5. How do I get a pass status?
etc.

**Comment by schlpbch** [comment]
As SBB we are also deeply interested in selling passes as Swiss Travel Products are an important product for tourists.

**Reply by mvdgraaff** [reply]
Shall we find the overlap between the passes and see of our way of working with osdm and passes matches the SBB way as well?
